identity-service/src/routes/solana.js

```diff
@@ -24,62 +24,73 @@ const isValidInstruction = (instr) => {
   return true
 }
 
-solanaRouter.post('/relay', authMiddleware, handleResponse(async (req, res, next) => {
-  const redis = req.app.get('redis')
-  const libs = req.app.get('audiusLibs')
-  let optimizelyClient
-  let socialProofRequiredToSend = true
-
-  let { instructions = [], skipPreflight, feePayerOverride } = req.body
-  try {
-    optimizelyClient = req.app.get('optimizelyClient')
-    socialProofRequiredToSend = getFeatureFlag(optimizelyClient, FEATURE_FLAGS.SOCIAL_PROOF_TO_SEND_AUDIO_ENABLED)
-  } catch (error) {
-    req.logger.error(`failed to retrieve optimizely feature flag for socialProofRequiredToSend: ${error}`)
-  }
-  if (socialProofRequiredToSend && isSendInstruction(instructions)) {
-    const userHasSocialProof = await doesUserHaveSocialProof(req.user)
-    if (!userHasSocialProof) {
-      let handle = req.user ? req.user.handle : ''
-      return errorResponseServerError(
-        `User ${handle} is missing social proof`,
-        { error: 'Missing social proof' }
-      )
+solanaRouter.post(
+  '/relay',
+  parameterizedAuthMiddleware({ shouldResponseBadRequest: false }),
+  handleResponse(async (req, res, next) => {
+    const redis = req.app.get('redis')
+    const libs = req.app.get('audiusLibs')
+    let optimizelyClient
+    let socialProofRequiredToSend = true
+
+    let { instructions = [], skipPreflight, feePayerOverride } = req.body
+    try {
+      optimizelyClient = req.app.get('optimizelyClient')
+      socialProofRequiredToSend = getFeatureFlag(optimizelyClient, FEATURE_FLAGS.SOCIAL_PROOF_TO_SEND_AUDIO_ENABLED)
+    } catch (error) {
+      req.logger.error(`failed to retrieve optimizely feature flag for socialProofRequiredToSend: ${error}`)
     }
-  }
+    if (socialProofRequiredToSend && isSendInstruction(instructions)) {
+      if (!req.user) {
+        return errorResponseServerError(
+          `User has no auth record`,
+          { error: 'No auth record' }
+        )
+      }
+
+      const userHasSocialProof = await doesUserHaveSocialProof(req.user)
+      if (!userHasSocialProof) {
+        let handle = req.user ? req.user.handle : ''
```

Comment:
req.user should exist here because if it didn't then we would have returned in the if above on line 45.

Reply:
yea

Reply:
fixed!